docs/structured-streaming-state-data-source.md

```diff
@@ -264,13 +264,14 @@ The output schema will also be different from the normal output.
 </tr>
 </table>
 
-## State metadata source
+## State Metadata Source
 
 Before querying the state from existing checkpoint via state data source, users would like to understand the information for the checkpoint, especially about state operator. This includes which operators and state store instances are available in the checkpoint, available range of batch IDs, etc.
 
 Structured Streaming provides a data source named "State metadata source" to provide the state-related metadata information from the checkpoint.
 
 Note: The metadata is constructed when the streaming query is running with Spark 4.0+. The existing checkpoint which has been running with lower Spark version does not have the metadata and will be unable to query/use with this metadata source. It is required to run the streaming query pointing the existing checkpoint in Spark 4.0+ to construct the metadata before querying.
+For operators using the operator metadata format version 1, the metadata is written once and does not change. For operators using metadata format version 2, this metadata can change and can be queried by providing the relevant batchId.
```

Comment:
We don't expect users to know about the operator metadata format, which operator uses the version 1 and which operator doesn't.

Flipping the coin, would there be a practice which users can simply follow without knowing the details? Shall we just guide users to always specify the batch ID if they want to know the metadata in "point of time", regardless of format version? My understanding is that the outcome should be the same.

Reply:
Done